ResearchKit/Common/ORKScaleSlider.m

```diff
@@ -60,17 +63,78 @@ - (id)initWithFrame:(CGRect)frame
         self.showThumb = NO;
         
         _axLastOutputTime = 0;
+        _thumbImageNeedsTransformUpdate = NO;
     }
     return self;
 }
 
-- (void)setShowThumb:(BOOL)showThumb
-{
+- (void)setShowThumb:(BOOL)showThumb {
     _showThumb = showThumb;
 }
 
-- (void)sliderTouched:(UIGestureRecognizer *)gesture
-{
+- (void)setVertical:(BOOL)vertical {
+    if (vertical != _vertical) {
+        _vertical = vertical;
+        self.transform = _vertical ? CGAffineTransformMakeRotation(-M_PI_2) : CGAffineTransformIdentity;
+        _thumbImageNeedsTransformUpdate = YES;
+    }
+}
+
+- (UIView *)thumbImageSubview {
+    UIView *thumbImageSubview = nil;
+    CGRect trackRect = [self trackRectForBounds:self.bounds];
+    CGRect thumbRect = [self thumbRectForBounds:self.bounds trackRect:trackRect value:self.value];
+    for (UIView *subview in self.subviews) {
+        if (CGRectEqualToRect(thumbRect, subview.frame)) {
+            thumbImageSubview = subview;
```

Comment:
This is ugly and error-prone. It would be better to have and explicitly set a thumb image asset. I'm willing to accept in the short term, though. Please add a comment to this effect and open an issue for fixing in future.